### ISSUE TITLE
FileEntity ACL

### ISSUE THREAD

**Issue by ARMmaster17** [CS4850Group5A/VirtuDoc#148]
# Summary
As implemented in #147, FileEntity records will be accessible by anyone. Need to implement an "ACL-style" attribute on all files so that only the patient (owner) of the documents and doctors associated with that patient can view those documents.

Blocked by #147

# Implementation
This is a proposal of how to implement this feature, and it is up to the ticket assignee to use their judgement to decide the final implementation.
- [x] Add a `owner` field to the `FileEntity` class that references a `UserAccount` instance.
- [x] Modify the records controller to only return file references that have the currently logged-in user marked as the owner.
- [ ] Modify the records controller so that if a doctor is logged in, only files belong to patients who have at least one appointment with that doctor will show up in the results (may need to be implemented in a separate issue as a helper method in another service class).
